#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/MattePaint.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0944
  position -1.4128506524952576e-05 -1.4128546475141566e-05 5.003411520814248
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 1.5 1.5
  wallHeight 0.5
  wallAppearance MattePaint {
    baseColor 0 0 0
  }
}
DEF YellowCylinder Solid {
  translation -0.625 0.625 0.14925008040921806
  rotation -5.989537882062313e-18 1.528979375874396e-17 1 1.5707963267948961
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.25
        radius 0.1016
      }
    }
  ]
  name "Goal"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  recognitionColors [
    1 1 0
  ]
}
DEF epuck E-puck {
  translation 0 0 0
  rotation 0 0 1 1.5708000060964915
  controller "Task1"
  supervisor TRUE
  turretSlot [
    DEF Front DistanceSensor {
      translation 0.045 0 -0.011
      name "front distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Right DistanceSensor {
      translation 0.01 -0.035 -0.011
      rotation 0 0 1 -1.5707853071795865
      name "right distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Rear DistanceSensor {
      translation -0.025 0 -0.011
      rotation 0 0 1 3.1415
      name "rear distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Left DistanceSensor {
      translation 0.01 0.035 -0.011
      rotation 0 0 1 1.5708
      name "left distance sensor"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    InertialUnit {
      rotation -1 0 0 -5.307179586466759e-06
    }
    Camera {
      translation 0.043 0 -0.015
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
